Skylark/Dispatch.cpp

```diff
@@ -502,18 +502,34 @@ void Http::Finalize()
         }
         else {
             r <<
-                code << ' ' << code_text << "\r\n" 
-                "Date: " <<  WwwFormat(GetUtcTime()) << "\r\n" 
-                "Content-Length: " << response.GetCount() << "\r\n" 
-                "Content-Type: " << content_type << "\r\n";
+                code << ' ' << code_text << "\r\n"
+                "Date: " <<  WwwFormat(GetUtcTime()) << "\r\n";
+                if(responseStream){
+                    r << "Content-Length: " << responseStream.GetSize() << "\r\n";
+                }else{
+					r << "Content-Length: " << response.GetCount() << "\r\n";
+                }
+                r << "Content-Type: " << content_type << "\r\n";
+                
             for(int i = 0; i < headers.GetCount(); i++)
                 r << headers.GetKey(i) << ": " << headers[i] << "\r\n";
             for(int i = 0; i < cookies.GetCount(); i++)
                 r << cookies[i];
         }
         r << "\r\n";
         rsocket->PutAll(r);
-        rsocket->PutAll(response);
+        if(responseStream){
+            Upp::String fileData = "";
```

Comment:
We are in Upp namespane, remove Upp::
String is initialized to empty one, remove = ""

Reply:
I'm noticing that change is not part of  the pull request. I have miss used my branche. However since I did a post about it : https://www.ultimatepp.org/forums/index.php?t=msg&goto=57568&#msg_57568 I will correct it